src/rule.js

```diff
@@ -49,6 +53,18 @@ class Rule extends EventEmitter {
     return this
   }
 
+  /**
+   * Sets the name of the rule
+   * @param {string} name - only non-empty strings are allowed
+   */
+  setName (name) {
+    if (!name || typeof name !== 'string') {
```

Comment:
```suggestion
    if (!name) {
```

@MattCopenhaver I'm wondering if there might be some benefit in allowing this field to be multiple types. For example, if a consumer wanted to set `name` to be an js object to enable multiple properties, e.g.
```js
rule.setName({
 id: '1234',
 type: 'user'
})
```

or in the case of multiple identifiers:
```js
rule.setName({
 pk: 1234,
 uuid: 'f6f14d07-04a4-4db1-9936-75b61069f25b'
})
```

definitely both are edge cases, but if the functionality comes for 'free' by being less strict, I don't see a down side to enabling it.

Reply:
Good suggestion.  I'll make this change.

Reply:
I went back and forth on the `id`/`name`/? thing as well. Honestly I think we should just stick with `name` - like you say the uniqueness isn't being enforced by the library, and these additional cases using objects will be an edge case.